src/components/tab-group/tab-group.ts

```diff
@@ -0,0 +1,120 @@
+import {Component, Input, ViewChildren, NgZone} from '@angular/core';
+import {QueryList} from '@angular/core';
+import {ContentChildren} from '@angular/core';
+import {PortalHostDirective} from '../../core/portal/portal-directives';
+import {MdTabLabel} from './tab-label';
+import {MdTabContent} from './tab-content';
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+
+/**
+ * Used to generate unique ID's for each tab component
+ * @type {number}
+ */
+let nextId = 0;
+
+@Component({
+  selector: 'md-tab-group',
+  templateUrl: './components/tab-group/tab-group.html',
+  styleUrls: ['./components/tab-group/tab-group.css'],
+  directives: [PortalHostDirective, MdTabLabelWrapper, MdInkBar],
+})
+export class MdTabGroup {
+  @ContentChildren(MdTabLabel) labels: QueryList<MdTabLabel>;
+  @ContentChildren(MdTabContent) contents: QueryList<MdTabContent>;
+  @ViewChildren(MdTabLabelWrapper) labelWrappers: QueryList<MdTabLabelWrapper>;
+  @ViewChildren(MdInkBar) inkBar: QueryList<MdInkBar>;
```

Comment:
You can use `@ViewChild` instead of `@ViewChildren` to just get a single instance.
```ts
 @ViewChild(MdInkBar) inkBar: MdInkBar;
```